### ISSUE TITLE
glome-login leaving an extra newline in the input buffer before returning?

### ISSUE THREAD

**Issue by delroth** [google/glome#116]
A tiny nit I've had with glome-login for a while (this behavior has been here for 1+ year): after I enter my authentication code and I get logged in, it seems like there is a spurious newline still in the input buffer which causes my shell to give me two prompts:

```
smol login: root
Obtain the one-time authorization code from:
<snip>
> <snip>
Authorization code: OK

[root@smol:~]# 

[root@smol:~]# 
```

**Comment by vvidic** [comment]
I noticed the same thing :)

It seems the size of the input buffer is enough to take the full auth code but not the newline (which is than left for the shell to read). If a shorter auth code is used than there is no problem. I can try to make a fix once #111 lands...

**Comment by vvidic** [comment]
Long auth code can also execute commands :)

```
Debian GNU/Linux 11 d06e5bd40062 -

d06e5bd40062 login: root
Obtain the one-time authorization code from:
https://glome.example.com/v1/AXIe9LWFjulCThtZH2SjfpLsHOhQ4DuGCbtLXv3YOUgP/d06e5bd40062/shell/root/
> QjVcpSm0vnupQLl3lAXtsl-jNGFQJZU40DLw0c1emGs=ls -l /
Authorization code: OK
Linux d06e5bd40062 5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
root@d06e5bd40062:~# ls -l /
total 64
drwxr-xr-x   2 root   root    4096 Jan 31 22:40 bin
drwxr-xr-x   2 root   root    4096 Dec 11 17:25 boot
drwxr-xr-x   5 root   root     340 Feb  3 22:09 dev
drwxr-xr-x  35 root   root    4096 Feb  3 22:09 etc
drwxr-xr-x   2 root   root    4096 Dec 11 17:25 home
drwxr-xr-x   9 root   root    4096 Jan 31 22:40 lib
...
```
